### ISSUE TITLE
unable to write incoming logs to opensearch when using otel input

### ISSUE THREAD

**Issue by ArgTang** [fluent/fluent-bit#7191]
## Bug Report

**Describe the bug**
Hello 
I am trying to use fluentbit as a opensearch output plugin for the open telemetry collector, as it does not have its own. I am only trying to write logs to opensearch, but i am unable to make it work. I have other fluentbit instances writing logs to opensearch without any issues. I dont have any parsing or filters in the pipeline for now. 

**To Reproduce**
helm chart: 0.24.0
image: 2.0.9
config:
````yaml
  kind: Deployment
  logLevel: debug 

  extraPorts:
    - port: 4318
      containerPort: 4318
      protocol: TCP
      name: otel

  inputs: |
    [INPUT]
        name opentelemetry
        listen 0.0.0.0
        port 4318

  outputs: |
    [OUTPUT]
        name stdout # stdout prints the logmessages
        match *

    [OUTPUT]
        Name opensearch 
        Match *
        Host opensearch-cluster-master.logging
        Port 9200
        Suppress_Type_Name On
        Trace_Error On
        Trace_Output On
        Replace_Dots On
        Index my_test_index
````

logs from fluentbit:
````log
[2023/04/14 12:52:51] [debug] [opensearch:opensearch.1] created event channels: read=34 write=35
[2023/04/14 12:52:51] [debug] [output:opensearch:opensearch.1] host=opensearch-cluster-master.logging port=9200 uri=/_bulk index=my_test_index type=_doc
[2023/04/14 12:52:51] [debug] [router] match rule http.0:stdout.0
[2023/04/14 12:52:51] [debug] [router] match rule http.0:opensearch.1
[2023/04/14 12:52:51] [debug] [router] match rule opentelemetry.1:stdout.0
[2023/04/14 12:52:51] [debug] [router] match rule opentelemetry.1:opensearch.1
[2023/04/14 12:52:51] [ info] [http_server] listen iface=0.0.0.0 tcp_port=2020
[2023/04/14 12:52:51] [ info] [sp] stream processor started
[2023/04/14 12:52:51] [ info] [output:stdout:stdout.0] worker #0 started
[2023/04/14 12:53:23] [debug] [input chunk] update output instances with new chunk size diff=73
[2023/04/14 12:53:23] [debug] [task] created task=0x7f20f00541c0 id=0 OK
[2023/04/14 12:53:23] [debug] [output:stdout:stdout.0] task_id=0 assigned to thread #0
[0] v1_logs: [1681476803.007763352, "this is a logmessagethis is a logmessagethis is a logmessage"]
[2023/04/14 12:53:23] [debug] [out flush] cb_destroy coro_id=0
[2023/04/14 12:53:23] [debug] [out flush] cb_destroy coro_id=0
[2023/04/14 12:53:23] [debug] [task] destroy task=0x7f20f00541c0 (task_id=0)
````

There is no indication that opensearch ever receives a blk request in this configuration. the index is never created, no logs of any requests in the elasticsearch cluster. 

**Expected behavior**
i would expect otel messages to be inserted into opensearch indice. Any help appriciated

**Your Environment**

* Version used: 2.0.9 (opensearch 2.4.1, otel collector v0.75.0)
* Configuration: see above
* Environment name and version (e.g. Kubernetes? What version?): running in a kubernetes pod receiving traffic from otel collector
* Server type and version: 1.24.9
* Filters and plugins: none

**Additional context**

**Comment by ArgTang** [comment]
updated to fluentbit 2.0.11, still no luck, although some more loggmessages has been added
`````logs
[2023/04/14 13:18:54] [debug] [upstream] KA connection #58 to opensearch-cluster-master.logging:9200 has been assigned (recycled)
[2023/04/14 13:18:54] [debug] [upstream] KA connection #58 to opensearch-cluster-master.logging:9200 is now available
[2023/04/14 13:18:54] [debug] [out flush] cb_destroy coro_id=13
[2023/04/14 13:18:54] [debug] [task] destroy task=0x7fc173a5b150 (task_id=0)
[2023/04/14 13:18:54] [debug] [input chunk] update output instances with new chunk size diff=73
[2023/04/14 13:18:55] [debug] [task] created task=0x7fc173a5b150 id=0 OK
[2023/04/14 13:18:55] [debug] [output:stdout:stdout.0] task_id=0 assigned to thread #0
[0] v1_logs: [1681478334.695862073, "this is a logmessagethis is a logmessagethis is a logmessage"]
[2023/04/14 13:18:55] [debug] [out flush] cb_destroy coro_id=14
[2023/04/14 13:18:55] [debug] [upstream] KA connection #58 to opensearch-cluster-master.logging:9200 has been assigned (recycled)
[2023/04/14 13:18:55] [debug] [upstream] KA connection #58 to opensearch-cluster-master.logging:9200 is now available
[2023/04/14 13:18:55] [debug] [out flush] cb_destroy coro_id=14
[2023/04/14 13:18:55] [debug] [task] destroy task=0x7fc173a5b150 (task_id=0)
````

**Comment by ArgTang** [comment]
Update: not able to get it working. switched to vector.dev, now we insert logs into opensearch from an oltp input

**Comment by yunyanng** [comment]
Same problem for elasticsearch output when using otel input.

If u add `raw_traces` for the otel input as such:
```
[INPUT]
    name opentelemetry
    listen 0.0.0.0
    port 4318
    raw_traces on
```
the protobuf **encoded** payload from otel will be forwarded to the output. We need a way to **decode** the payload before forwarding to output.

**Comment by leonardo-albertovich** [comment]
Could you please verify if this issue is still present in fluent-bit 2.1?

**Comment by yunyanng** [comment]
Tested with fluent-bit 2.12, same results.

**Comment by yunyanng** [comment]
Just to be clear, my output is elasticsearch, not opensearch.

**Comment by leonardo-albertovich** [comment]
This issue should be fixed by PR #3282 which will be included in the upcoming release.

**Comment by yunyanng** [comment]
May I know which release? I don't find it in 2.1.3 which was released yesterday.

**Comment by leonardo-albertovich** [comment]
That's because this error is not present in fluent-bit 2.1 so it was only fixed in fluent-bit 2.0 which is scheduled to be released on Monday.

If you are having trouble with fluent-bit 2.1 please create a new issue so we can properly investigate them.
